fix: NetworkPrefabs container's elements invalidated in the NetworkManager after relaunching Unity Project

NoelStephensUnity opened the pull request:
**This resolves the following issue:**
Whilst upgrading Boss Room to MLAPI develop, we encountered an issue with the Network Manager and its NetworkPrefabs container, where if we delete any of the prefabs or variants that were added to the NetworkPrefabs list, then save (apply changes to) the NetworkingManager prefab, then shut down Unity and load it up again, the NetworkPrefabs list will be invalidated, most of the elements will have a null value.
[Github Issue 904](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/904)

**Additional Notes:**
This issue only arises if you make changes to a prefab instance and then apply those changes to the source prefab using Overrides->Apply All.  This only applies to the NetworkPrefabs list generation while in the editor as this list is rebuilt during runtime and the only time you would want to generate the list while in the editor is if the currently opened scene is dirty and the asset database is not being updated.

NoelStephensUnity committed:
fix

Resolves the following github issue:
NetworkPrefabs container's elements invalidated in the NetworkManager after relaunching Unity Project #904
https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/904

Checking to make sure the scene is dirty and that the asset database is not currently updating resolves this issue.

NoelStephensUnity committed:
style

Updated the comment for better clarity and assuring we have at least two commits in order to pass Yamato requirement of >1 commit

MFatihMAR commented:
I'd like to defer this one to @Cosmin-B

0xFA11 commented on com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs:
```diff
@@ -261,54 +261,58 @@ private void OnValidate()
-
-            // Check network prefabs and assign to dictionary for quick look up
-            for (int i = 0; i < NetworkConfig.NetworkPrefabs.Count; i++)
+            // If the scene is dirty and the asset database is not currently updating then we can update NetworkPrefab information
+            if (activeScene.isDirty && !UnityEditor.EditorApplication.isUpdating)
```
IDK why Git is confused about the diff but this part is the only diff we actually have — and it looks good to me! 🚀

NoelStephensUnity replied:
I believe it is because all of the indentions and line numbers changed when I enclosed all of the Network Prefab list stuff within that if statement.

Cosmin-B replied:
It looks good to me, the only thing I am slightly worried about, is whether we can use a conditional early return/guard instead of enclosing all the code below, it makes it less readable given the Identations.

NoelStephensUnity committed:
refactor

Updated logic to exit OnValidate as opposed to wrapping the NetworkPrefab related code for easier readability.
